metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -200,13 +200,50 @@ class ScalametaLanguageServer(
   override def documentSymbols(
       td: TextDocumentIdentifier
   ): Seq[SymbolInformation] = {
-    val path = Uri.toPath(td.uri).get
-    symbol.documentSymbols(path.toRelative(cwd)).map {
-      case (position, denotation @ Denotation(_, name, signature, _)) =>
-        val location = path.toLocation(position)
-        val kind = denotation.symbolKind
-        SymbolInformation(name, kind, location, Some(signature))
+    import scala.meta._
+
+    // For a given node returns its closest ancestor which is a definition, declaration or a package object
+    // NOTE: package is not considered a wrapping definition, but it could be (a subject to discuss)
+    def wrappingDefinition(t: Tree): Option[Tree] = {
+      if (
+        t.is[Defn] ||
+        t.is[Decl] ||
+        t.is[Pkg.Object]
+      ) Some(t)
+      else t.parent.flatMap(wrappingDefinition)
+    }
+
+    // This is needed only to unfold full package names
+    def qualifiedName(t: Tree): Option[String] = t match {
+      case Term.Name(name) =>
+        Some(name)
+      case Term.Select(qual, name) =>
+        qualifiedName(qual).map { prefix => s"${prefix}.${name}" }
+      case Pkg(sel: Term.Select, _) =>
+        qualifiedName(sel)
+      case m: Member =>
```

Comment:
Glad you found this! Took me a while to learn about it :)

Reply:
What do you mean? 😅

Reply:
I wasn't aware of `Member` until quite recently, always found it frustrating to handle all val/val/def/class/trait/object cases

Reply:
Ah, yes, it's very helpful. I just searched for whatever with `name`, when I wanted to match it with symbols from the indexer 😄